src/models/Observation.js

```diff
@@ -108,19 +147,34 @@ class Observation {
     primaryKey: "uuid",
     properties: {
       uuid: "string",
+      captive_flag: "bool?",
       comments: "Comment[]",
       created_at: { type: "string?", mapTo: "createdAt" },
       description: "string?",
+      geoprivacy: "string?",
+      id: "int?",
       identifications: "Identification[]",
       latitude: "double?",
       longitude: "double?",
       observationPhotos: "ObservationPhoto[]",
-      observationSounds: "ObservationSound[]",
+      observationSound: "ObservationSound?",
+      observed_on_string: "string?",
+      owners_identification_from_vision_requested: "bool?",
+      species_guess: "string?",
       place_guess: { type: "string?", mapTo: "placeGuess" },
+      positional_accuracy: "double?",
       quality_grade: { type: "string?", mapTo: "qualityGrade" },
       taxon: "Taxon?",
       time_observed_at: { type: "string?", mapTo: "timeObservedAt" },
+      time_zone: "string?",
+      timeSynced: "date?",
```

Comment:
I think I'm still getting mixed up on the difference between synced at, updated at, observed at, created at, and where we need to use each of them respectively.

I've been thinking of `time_observed_at` and `created_at` as the times coming back from the server, `timeSynced` as the time when data is downloaded from the server to the local device, and `timeUpdatedLocally` as the time when an observation gets saved locally for future upload. Does that sound right? Am I missing any other times that I need here? Or are there any I could combine? `created_at` and `timeUpdatedLocally` seem like basically the same thing to me, except that the source of truth is the server vs. the local device.

Reply:
These are the relevant date or time fields from the server

* `created_at`: timestamp of when observation was created on the server; not editable
* `updated_at`: timestamp of when the observation was updated on the server; not editable
* `observed_on`: date when the observer observed the organism; user-editable, but only by changing `observed_on_string`
* `time_observed_at`: datetime when the observer observed the organism; user-editable, but only by changing `observed_on_string` (the reason we have both is because an observation can specify a date without specifying a time)

Locally, you probably want to track

* `_created_at`: datetime the observation was created on the device
* `_updated_at`: datetime the observation was updated on the device; you probably want to track this because you want to know what records were updated since the last sync, but you don't want to sort by this field, otherwise your list of observations will change around every time you edit an observation
* `_synced_at`: datetime the observation was last synced with the server, useful for determining which observations have never been uploaded and which ones have been edited locally and need to be updated on the server